package zlog_ceph_proto;

option optimize_for = SPEED;

message StripeParams {
  required uint32 entry_size = 1;
  required uint32 stripe_width = 2;
  required uint32 entries_per_object = 3;
  required uint64 object_id = 4;
}

message ObjectMeta {
  required StripeParams params = 1;
}

message InitOp {
  required StripeParams params = 1;
}

message ReadOp {
  required uint64 position = 2;
  enum Status {
    OK = 0;
    UNWRITTEN = 1;
    INVALID = 2;
  }
}

message WriteOp {
  required uint64 position = 2;
  required bytes data = 3;
}

message InvalidateOp {
  required uint64 position = 1;
  required bool force = 2;
}
